lib/iris/_lazy_data.py

```diff
@@ -93,6 +93,8 @@ def as_lazy_data(data, chunks=_MAX_CHUNK_SIZE):
         The input array converted to a dask array.
 
     """
+    if isinstance(data, ma.core.MaskedConstant):
+        data = ma.masked_array([data], mask=data.mask)
```

Comment:
Sorry my last comment makes no sense, that's actually relevant to the tests.

Can we change this line to be
`data = ma.masked_array(data, mask=data.mask)`
?